File: MSc AI/Assignments/Intro to computing/Assingment scripts/clean_names_modified.py
# clean_names.py

# Function to process a list of names: remove duplicates, sort alphabetically, and convert to uppercase
def process_names(names):
    names = [name.upper() for name in names]  # Convert all names to uppercase for consistency
    unique_names = list(set(names))  # removes duplicates but doesn't preserve order
    unique_names.sort()  # sorts alphabetically
    return unique_names 

# Test list with duplicates and inconsistent capitalisation
test_names = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"]

print("Original list:", test_names)
print("Processed list:", process_names(test_names))

File: MSc AI/Assignments/Intro to computing/Assingment scripts/cleanNames_modified.js
// cleanNames.js

// Function to process a list of names: remove duplicates, sort alphabetically, and convert to uppercase
function processNames(names) {
    let namesNew = names.map((name) => name.toUpperCase()); // Convert all names to uppercase for consistency
    let uniqueNames = [...new Set(namesNew)]; // removes duplicates
    uniqueNames.sort(); // sorts alphabetically
    return uniqueNames;
}

// Test list with duplicates and inconsistent capitalisation
const testNames = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"];

console.log("Original list:", testNames);
console.log("Processed list:", processNames(testNames));